### DISCUSSION TITLE
18/7/22: The Forum Helpers Dashboard Update

### DISCUSSION THREAD

**Discussion by Accio1** [theforumhelpers/theforumhelpers.github.io#329] [Announcements]
The Forum Helpers Dashboard has received an update which allows curators to access the dashboard, as well as allowing people to change their bio on TFH website right from the dashboard! Please take the time to read the changelog and especially the things to note section below for important information about the dashboard and how to use it.

Dashboard Link: https://theforumhelpers.herokuapp.com/ 

### Changelog:
- Add a curator dashboard
    - When signing in, if someone is a curator, they will be directed to the curator dashboard. Otherwise, they will be directed to the manager dashboard.
    - Curators have the option to request a biography change, which will require approval by an admin before being displayed on TFH site.
- Add bio changing to the manager dashboard
    - Managers now have the option to request a biography change, which will require approval by an admin before being displayed on TFH site.
- Convert the list of Forum Helpers to a database
    - Instead of hardcoded JSON lists, the list of Forum Helpers is now stored in a database so that it can easily be updated.
    - Admins will be able to add new users or modify existing users through an admin dashboard.
        - Once an admin adds a user, they will immediately be able to log into the dashboard.
- Adds an admin dashboard
    - Admins have the ability to add users, modify users, and remove users. Additionally, admins approve or deny bio requests.
    - Current admins are @Accio1 and @leahcimto
- A variety of bug fixes

### Things to note:
- All submitted biographies **must** follow the Scratch Community Guidelines. Since bios need to be approved, inappropriate bios will be rejected by site admins and deliberately submitting an inappropriate biography will lead you to be removed from the dashboard.
- If you were recently accepted into TFH and you have not gained access to TFH Dashboard, please wait to be added by an admin. If you are not added within a week, please contact an admin.
- If you are having difficulties using the dashboard, do not hesitate to leave a comment in TFH studio requesting help.
- As of #327 being merged, TFH Dashboard database is connected to the main TFH site. This means that changes you make such as changing the application status will be reflected on the site. Abusing the dashboard to make unwarranted changes from the main site will result in your removal from the dashboard.
- If you notice a security vulnerability with the dashboard, please follow these steps:
    1. Do not abuse it - Don't deliberately take advantage of the vulnerability.
    2. Do not report it publicly - Don't share the vulnerability with others.
    3. Report it to @Accio1 - Report the vulnerability in the comments of one of my older Scratch projects. This way, it is unlikely that anyone else will see the report and I will be able to take action to fix the vulnerability.

@leahcimto

**Comment by penguinmoose** [comment]
@Accio1 @leahcimto

IMPORTANT! Please change the codeword back. I was kinda testing (sorry). Now that I know, I won't do it again unless the codeword needs to be changed. Please don't remove me from the dashboard.

Or if it's ok, I guess it is.

**Reply by Accio1** [reply]
@penguinmoose It's not really a big deal since it just picks from a list of about 50 animals. It should be changed about once a week or so anyway. Just be aware moving forward that this is the production version of the dashboard so any changes that you make are interacting with the real database and potentially the real website.

**Reply by penguinmoose** [reply]
Got it. I thought it would pop up a textbox for me to enter a word (in which case I would just have closed it because I didn't mean to actually change it).

**Comment by Weredime** [comment]
Great work! It is strange that https://theforumhelpers.herokuapp.com/api/users.php doesn't return a `Content-Type` of `application/json` though.

**Reply by Steve0Greatness** [reply]
I think it'd be pretty easy to do with `header("Content-Type:application/json")` in the PhP source code. Too bad it's not open source so that anybody can fix issues with the dashboard.

**Comment by Steve0Greatness** [comment]
Is it possible that at some point in the future there will be a public Git repository hosted on GitHub or GitLab? Or is that never going to happen?

**Reply by penguinmoose** [reply]
Probably, but maybe not in the near future (low priority according to the creator of the dashboard).

**Reply by aidan-crockett** [reply]
https://github.com/theforumhelpers/theforumhelpers.github.io/issues/297#issuecomment-914768233 
check the timestamp btw

**Reply by aidan-crockett** [reply]
i last asked this 10 months ago lol

**Reply by Steve0Greatness** [reply]
What's it with people thinking that open sourcing things it a "low priority?" It allows things to be more easily reported, and hopefully fixed faster(by anybody if you decide to allow PRs). It's just generally a good idea for most things. (replying to penguinmoose)

**Reply by aidan-crockett** [reply]
Have there been any major issues with it so far?

**Reply by Steve0Greatness** [reply]
No, it's just nice to have the ability to.

**Reply by penguinmoose** [reply]
@Steve0Greatness I have no idea why they think so. I agree that it is important for it to be open source (not low priority). It might be because the dashboard still has changes to be made (new features, bug fixes, etc).

**Reply by penguinmoose** [reply]
> [#297 (comment)](https://github.com/theforumhelpers/theforumhelpers.github.io/issues/297#issuecomment-914768233) check the timestamp btw

lol, it's 2022 and the source isn't even out yet

**Reply by Weredime** [reply]
Never going to happen since its PHP

**Reply by Accio1** [reply]
> > [#297 (comment)](https://github.com/theforumhelpers/theforumhelpers.github.io/issues/297#issuecomment-914768233) check the timestamp btw
> 
> lol, it's 2022 and the source isn't even out yet

@penguinmoose @gosoccerboy5 Please take into consideration that this is not the only thing I work on. Between school, work, extracurriculars and applying to university, this really wasn't at the top of my priority list. That timeline was predicted by me at the start of last school year before I got extremely busy and didn't have the opportunity to spend time working on the dashboard.

**Reply by aidan-crockett** [reply]
I understand that you're busy, and also that open-sourcing or exposing TFH dashboard is low priority. Sorry if I sounded rude because I didn't mean to

**Reply by penguinmoose** [reply]
@Accio1 Oh, sorry then. I understand.

**Reply by Steve0Greatness** [reply]
@Weredime What in the world do you mean by that? I've made an [open source program with PHP](/Steve0Greatness/War-of-the-hello-worlds).

@Accio1 if so, wouldn't it be better to make it open source so that other people can create pull requests, as well as adding a few other maintainers so that you aren't the soul person reviewing them?

**Reply by Weredime** [reply]
> @Weredime What in the world do you mean by that? I've made an [open source program with PHP](/Steve0Greatness/War-of-the-hello-worlds).

called a joke
